x-pack/plugins/spaces/public/management/edit_space/customize_space/customize_space_avatar.tsx

```diff
@@ -106,93 +118,192 @@ export class CustomizeSpaceAvatar extends Component<Props, State> {
+                    fullWidth
+                  />
+                </EuiFormRow>
+              </EuiFlexItem>
+            )}
+            <EuiFlexItem grow={false}>
+              <EuiSpacer size="m" />
+              <EuiSpacer size="xs" />
+              {space.avatarType === 'image' && space.imageUrl ? (
+                <Suspense fallback={<EuiLoadingSpinner />}>
+                  <LazySpaceAvatar
+                    space={{
+                      ...space,
+                      initials: undefined,
+                      name: undefined,
+                    }}
+                    size="l"
+                  />
+                </Suspense>
+              ) : space.avatarType !== 'image' && (space.name || space.initials) ? (
+                <Suspense fallback={<EuiLoadingSpinner />}>
+                  <LazySpaceAvatar
+                    space={{
+                      ...space,
+                      imageUrl: undefined,
+                    }}
+                    size="l"
+                  />
+                </Suspense>
+              ) : (
+                <EuiAvatar
+                  type="space"
+                  name="?"
+                  size="l"
+                  color={null}
+                  style={{
+                    background: euiThemeVars.euiColorLightShade,
+                    color: euiThemeVars.euiTextSubduedColor,
+                  }}
+                />
+              )}
+            </EuiFlexItem>
+          </EuiFlexGroup>
         </EuiFormRow>
-        <EuiSpacer size="m" />
-        {this.filePickerOrImage()}
       </form>
     );
   }
 
-  private removeImageUrl() {
-    this.props.onChange({
-      ...this.props.space,
-      imageUrl: '',
-    });
-  }
-
-  public filePickerOrImage() {
-    if (!this.props.space.imageUrl) {
-      return (
-        <EuiFormRow
-          label={i18n.translate('xpack.spaces.management.customizeSpaceAvatar.imageUrl', {
-            defaultMessage: 'Custom image',
-          })}
-        >
-          <EuiFilePicker
-            display="default"
-            data-test-subj="uploadCustomImageFile"
-            initialPromptText={i18n.translate(
-              'xpack.spaces.management.customizeSpaceAvatar.selectImageUrl',
-              {
-                defaultMessage: 'Select image file',
-              }
-            )}
-            onChange={this.onFileUpload}
-            accept={imageTypes.join(',')}
-          />
-        </EuiFormRow>
-      );
-    } else {
-      return (
-        <EuiFlexItem grow={true}>
-          <EuiButton onClick={() => this.removeImageUrl()} color="danger" iconType="trash">
-            {i18n.translate('xpack.spaces.management.customizeSpaceAvatar.removeImage', {
-              defaultMessage: 'Remove custom image',
-            })}
-          </EuiButton>
-        </EuiFlexItem>
-      );
-    }
-  }
-
   public initialsInputRef = (ref: HTMLInputElement) => {
```

Comment:
nit: we can remove this function as well as `onInitialsFocus` and `onInitialsBlur` now 🎉